File: backend/src/clases/dto/dto/create-task.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsDateString, IsOptional, IsString } from 'class-validator';

export class CreateTaskDataIn {
  @ApiProperty()
  @IsString()
  title: string;

  @ApiProperty()
  @IsString()
  classId: string;

  @ApiProperty()
  @IsString()
  topicId: string;

  @ApiProperty()
  @IsOptional()
  @IsString()
  description?: string;

  @ApiProperty()
  @IsOptional()
  @IsDateString()
  dueDate?: Date;

  @ApiProperty()
  @IsOptional()
  @IsString()
  score?: string;
}

File: fontend/lib/domain/dtos/create_task.dto.dart
import 'dart:io';

class CreateTaskDTO {
  String? title;
  String? description;
  String? topicId;
  String classId;
  int? score;
  DateTime? date;
  List<File>? files;

  CreateTaskDTO({
    this.title,
    this.description,
    this.topicId,
    required this.classId,
    this.score,
    this.date,
    this.files,
  });
  Map<String, dynamic> toMap() {
    return {
      'title': title,
      'classId': classId,
      'topicId': topicId,
      'description': description,
      'dueDate': date?.toIso8601String(),
      'score': score,
    };
  }
}
